### PR TITLE
Fix not equals

### PR DESCRIPTION
**Pull request by vasukalariya**
# Description
Summarize the change.
For the operator `NotEquals`, we cannot do a rangeCheck directly on minVal and maxVal, for example if a block has different minVal and maxVal, we cannot return that block as false, since there are records in that block which still match for operator `NotEquals`. The only case where we can reject the block is when all minVal, maxVal and lookupVal are same.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?
`http_status != 200 | stats count` was not giving any results before for benchmark dataset.
Now it works as expected.

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/query/metadata/metautils/metacheckers.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func doesUintPassRangeFilter(op utils.FilterOperator, lookupValue uint64, minVal
 	case utils.Equals:
 		return lookupValue >= minVal && lookupValue <= maxVal
 	case utils.NotEquals:
-		if lookupValue == minVal || lookupValue == maxVal {
+		if minVal == maxVal && lookupValue == minVal {
```
Comment (AndrewHess) [inline pkg/segment/query/metadata/metautils/metacheckers.go:109]: I think your change is correct, but not as restrictive as it could be. I think `utils.NotEquals` should be the same as `not utils.Equals`, which is
```
not (lookupValue >= minVal && lookupValue <= maxVal)
```
which is
```
lookupValue < minVal || lookupValue > maxVal
```
Am I missing something?

Comment (AndrewHess) [reply pkg/segment/query/metadata/metautils/metacheckers.go:109]: Also, can you check the other inequalities? They look odd to me but I'm not quite convinced if they're right/wrong. There's a basic unit test for this function, but it would be good to add some more tests for it to catch edge cases like the one you found

Comment (vasukalariya) [reply pkg/segment/query/metadata/metautils/metacheckers.go:109]: Wouldn't this condition `lookupValue < minVal || lookupValue > maxVal` give incorrect results for the case where lookupValue is 300, minVal is 200 and maxVal is 500. It would give `false` for that block even though there are records satisfying this condition.

Comment (AndrewHess) [reply pkg/segment/query/metadata/metautils/metacheckers.go:109]: Ah, I think I was misunderstanding exactly how this function is supposed to behave. Now I think this is the expected behavior:

Returns true if there exists an `actualValue` with `minValue <= actualValue <= maxValue` such that `actualValue op lookupValue` is true

Is that what these functions are supposed to do? Can you please add a comment to the code explaining what the function does, and then I'll approve this PR

Comment (vasukalariya) [reply pkg/segment/query/metadata/metautils/metacheckers.go:109]: Yes that is correct.

**Review by AndrewHess — APPROVED** [review]
